orchestrator/src/compression/squash.rs

```
@@ -0,0 +1,355 @@
+            let contract_existed = check_contract_existed_at_block(&provider, contract_addr, pre_range_block).await;
+
+            if contract_existed {
+                // Process storage entries only for an existing contract
+                let results: Vec<_> = stream::iter(storage_map)
+                    .map(|(key, value)| {
+                        let provider = provider.clone();
+                        async move {
+                            let pre_range_value =
+                                check_pre_range_storage_value(&provider, contract_addr, key, pre_range_block).await?;
+
+                            Ok::<_, JobError>((key, value, pre_range_value))
+                        }
+                    })
+                    .buffer_unordered(MAX_CONCURRENT_GET_STORAGE_AT_CALLS)
+                    .collect()
+                    .await;
+
+                // Process results
+                for result in results {
+                    let (key, value, pre_range_value) = result?;
+                    if pre_range_value != value {
+                        storage_entries.push(StorageEntry { key, value });
+                    }
+                }
+            } else {
+                // Contract didn't exist, filter non-zero values
+                for (key, value) in storage_map {
+                    if value != Felt::ZERO {
+                        storage_entries.push(StorageEntry { key, value });
+                    }
+                }
+            }
+        }
+    }
+
+    if !storage_entries.is_empty() {
+        Ok(Some(ContractStorageDiffItem { address: contract_addr, storage_entries }))
+    } else {
+        Ok(None)
+    }
+}
+
+pub async fn check_contract_existed_at_block(
+    provider: &Arc<JsonRpcClient<HttpTransport>>,
+    contract_address: Felt,
+    block_number: u64,
+) -> bool {
+    match get_class_hash_at(provider, block_number, contract_address).await {
+        Ok(_) => true,
+        Err(_) => false,
```

You should return false only if the error is ‘class not found’. Otherwise, you should return an error in all other cases.

---

Fixed